jupyter_client/manager.py

```diff
@@ -250,6 +240,8 @@ def start_kernel(self, **kw):
             # If kernel_cmd has been set manually, don't refer to a kernel spec
             # Environment variables from kernel spec are added to os.environ
             env.update(self.kernel_spec.env or {})
+        elif self.extra_env:
+            env.update(self.extra_env)
```

Comment:
KernelManager is rather complex because of the evolution of how we use it.

1. Originally it had a configurable kernel command line, which you could use to make it start a different kernel, before Jupyter (then IPython) was really aware of other kernels. But this changed it for every KernelManager in a process, so you couldn't mix kernels in one notebook server, or one Qt console.
2. Then with kernelspecs, it was meant to be given the name of a kernelspec, and it would retrieve argv and env from that. But we left the machinery for 1. in place so that people configuring `kernel_cmd` wouldn't find it broken as soon as they upgraded.
3. In this PR, I've reused the leftover machinery for 1. to allow kernel_cmd as an input, but not as a config option, and added `extra_env` to fit with this way of doing it. But I don't want to immediately break things for code using 2. This `elif` branch is for 3., and the `if` branch above is for 2. They shouldn't be mixed.